driver-sync/src/test/functional/com/mongodb/client/Fixture.java

```diff
@@ -44,10 +43,23 @@ private Fixture() {
     }
 
     public static synchronized MongoClient getMongoClient() {
-        if (mongoClient == null) {
-            mongoClient = MongoClients.create(getMongoClientSettings());
-            Runtime.getRuntime().addShutdownHook(new ShutdownHook());
+        if (mongoClient != null) {
+            return mongoClient;
         }
+        MongoClientSettings mongoClientSettings = getMongoClientSettings();
+        mongoClient = MongoClients.create(mongoClientSettings);
+        Runtime.getRuntime().addShutdownHook(new Thread(() -> {
+            synchronized (Fixture.class) {
+                if (mongoClient == null) {
+                    return;
+                }
+                if (defaultDatabase != null) {
+                    defaultDatabase.drop();
```

Comment:
This refactoring just moves the logic close to where it is used.

I do not really understand why we use a shutdownHook here, when elsewhere we rely on `@AfterEach/All` to clear resources. Perhaps we should either be using something that registers resources for shutdown with a shutdown hook, or just not using a shutdown hook at all.